### PR TITLE
Factors/reactivate keys

### PR DESCRIPTION
**Pull request by Fucci-Mateo**

### CODE REVIEW

File: workers/factors/automation/data.go [hunk 1]
```diff
@@ -56,6 +70,11 @@ func (worker *Worker) FetchData(ctx context.Context) (map[string]*FactorReport,
 		return nil, nil, err
 	}
 
+	worker.InactiveKeys, err = worker.FetchInactiveKeys(ctx)
```
Comment (finkiel) [inline workers/factors/automation/data.go:73]: please add some log if the function returns an err - so we'll be able to monitor it
if it suppose to happen lots of time so don't
right now, you are throwing an err without any data

File: workers/factors/automation/data.go [hunk 2]
```diff
@@ -260,10 +279,47 @@ func FetchAutomationSetup() (map[string]*DomainSetup, error) {
...
 	if gppTarget != 0 {
 		gppTarget /= 100
 	}
 	return gppTarget
 }
+
+func (worker *Worker) FetchGppTargetDefault() (float64, error) {
+	GppTargetString, err := config.FetchConfigValues([]string{"factor-automation:gpp-target"})
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to fetch gpp target value from API")
```
Comment (finkiel) [inline workers/factors/automation/data.go:314]: if you can add the string that you're trying to fetch or the factor for which you don't have gpp_target

File: workers/factors/automation/data.go [hunk 3]
```diff
@@ -260,10 +279,47 @@ func FetchAutomationSetup() (map[string]*DomainSetup, error) {
...
+	}
+
+	GppTargetFloat, err := strconv.ParseFloat(GppTargetString["factor-automation:gpp-target"], 64)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to convert GppTarget value to float")
```
Comment (finkiel) [inline workers/factors/automation/data.go:319]: again, without any data we won't be able to monitor in the future if we'll have any bug
add the gppTarget value that you are trying to convert

File: workers/factors/automation/data.go [hunk 4]
```diff
@@ -260,10 +279,47 @@ func FetchAutomationSetup() (map[string]*DomainSetup, error) {
 	return domainsMap, nil
 }
 
+// Fetch inactive keys from postgres
```
Comment (finkiel) [inline workers/factors/automation/data.go:282]: no need in comment - the name of the function is clear enough

File: workers/factors/automation/data.go [hunk 5]
```diff
@@ -36,6 +39,17 @@ WHERE timestamp >= '%s'
   AND domain in('%s')
 GROUP BY 1, 2, 3, 4, 5`
 
+var InactiveKeysQuery = `SELECT *
+FROM (SELECT publisher,
```
Comment (finkiel) [inline workers/factors/automation/data.go:43]: it's a stupid comment but you don't need to break each work to a new line:
`SELECT *
         FROM (SELECT publisher,domain, country, device,SUM(CASE WHEN new_factor >= %f THEN 1 ELSE 0 
         END) AS positive_cases)
FROM ......

File: workers/factors/automation/data.go [hunk 6]
```diff
@@ -36,6 +39,17 @@ WHERE timestamp >= '%s'
   AND domain in('%s')
 GROUP BY 1, 2, 3, 4, 5`
 
+var InactiveKeysQuery = `SELECT *
```
Comment (finkiel) [inline workers/factors/automation/data.go:42]: as long as your'e not using the query elsewhere you should name it inactiveKeyQuery (lower I)

File: workers/factors/automation/data.go (hunk 1 above)
Comment (Fucci-Mateo) [reply workers/factors/automation/data.go:73]: tecnically it has only one error posibility (line  292) which is: "failed to fetch the query", so I return the error on the FetchInactiveKeys() -> gets passed to the FetchData() function -> gets passed to the wordker.Do() function -> It gets logged and alerted

File: workers/factors/automation/data.go (hunk 6 above)
Comment (Fucci-Mateo) [reply workers/factors/automation/data.go:42]: done

File: workers/factors/automation/data.go (hunk 4 above)
Comment (Fucci-Mateo) [reply workers/factors/automation/data.go:282]: done

**Review by finkiel — APPROVED** [review]

**Pull request merged**